src/arrayMethodSort.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let counter;
+
+    do {
+      counter = 0;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const aString = String(this[i]);
+        const bString = String(this[i + 1]);
+        const a = this[i];
+        const b = this[i + 1];
+
+        if (compareFunction === undefined && aString > bString) {
```

Comment:
Think you can modify this `if else` statement into one to be:

`if (compareFunction === undefined && aString > bString || compareFunction !== undefined && compareFunction(a, b) > 0)`

They doing the same operations

Reply:
`if ((compareFunction === undefined && aString > bString) || (compareFunction !== undefined && compareFunction(a, b) > 0))
`

how about this one?

Reply:
still the same